pkg/network/driver_interface.go

```diff
@@ -418,3 +570,32 @@ func newDDAPIFilter(direction, layer C.uint64_t, ifaceIndex int, isIPV4 bool) C.
 
 	return fd
 }
+
+// prepare N read buffers
+// and return the IoCompletionPort that will be used to coordinate reads.
+// danger: even though all reads will reference the returned iocp, buffers must be in-scope as long
+// as reads are happening. Otherwise, the memory the kernel is writing to will be written to memory reclaimed
+// by the GC
+func prepareCompletionBuffers(h windows.Handle, count int) (iocp windows.Handle, buffers []*_readbuffer, err error) {
+	iocp, err = windows.CreateIoCompletionPort(h, windows.Handle(0), 0, 0)
+	if err != nil {
+		return windows.Handle(0), nil, errors.Wrap(err, "error creating IO completion port")
+	}
+
+	buffers = make([]*_readbuffer, count)
+	for i := 0; i < count; i++ {
+		buf, err := allocateReadBuffer()
```

Comment:
This may expose my lack of `go` expertise; but is allocating using `cgo malloc` necessary?  I think we can use the regular go memory allocation, and as long as we hold a reference so the garbage collector doesn't take it back, we're OK.
As far as correctness it doesn't matter, but simplifies needing to go out to `cgo`, have separate 386/x64 files, etc.

Reply:
Unfortunately, it is necessary since the C code hangs on to the pointer. From https://golang.org/cmd/cgo/#hdr-Passing_pointers
> C code may not keep a copy of a Go pointer after the call returns

Reply:
I originally hadn't used malloc to allocate, but that caused memory corruption to happen. The theory is:

* readBuffer was allocated on the go heap
* we passed the read buffer to ReadFile()
* go relocated the readBuffers on the go heap
* ReadFile wrote to the wrong location
* the runtime crashed